### ISSUE TITLE
Migrate to the new linode_v4 module

### ISSUE THREAD

**Issue by decentral1se** [ansible/molecule#1678]
# Issue Type
- Feature request

# Issue Description

Since https://github.com/ansible/molecule/pull/1458 is now merged, there are 2 remaining tasks to do.

* Since https://github.com/ansible/ansible/pull/46151 is merged (to arrive in 2.8), we can drop the deprecated linode module being used here. That can't happen until 2.8 is out but should happen before https://github.com/ansible/ansible/issues/51060 happens :) AFAIK, there is already enough functionality in the new module to make this driver work.

* ~~Currently, the functional tests for the linode module are skipped. There is a nice recommendation to use the linode_v4 client fixtures (seen in https://github.com/linode/linode_api4-python/tree/master/test/fixtures) to mock the network connection necessary for running the functional tests. Any test server developed for this work would be of general use to other test drivers who want to mock the network as well.~~
  * Better plan, see: https://github.com/ansible/molecule/issues/1678#issuecomment-493737894.

This work mostly waits on the release of 2.8.

**Comment by decentral1se** [comment]
I've started a discussion on `#ansible-linode` to see if we can get account access for firing up Linode boxes temporarily for the CI testing. Something like we are getting for Hetzner Cloud (see https://github.com/ansible/molecule/pull/1811#issuecomment-487449174).

**Comment by decentral1se** [comment]
FYI, the discussion is underway ... tokens / CI/CD account access is being arranged :rocket:

**Comment by decentral1se** [comment]
I won't be having to follow up on this in the end as I'm passing the torch on the Linode implementation stuff I was doing. Going to close off (please re-open on another ticket if needed, I use my issues as a TODO list).

**Comment by decentral1se** [comment]
OK, I'll try to take a stab at the new module. No functional testing for now but that can come later on when the plugin architecture comes into place. Since we're aiming for a 3.0 release, I won't handle backwards support for the new modules.
